From 9953e604ceb7c17a82cd6b457a4bbb1438d2dc75 Mon Sep 17 00:00:00 2001
From: Alex Bee <knaerzche@gmail.com>
Date: Sun, 15 Dec 2024 12:53:18 +0100
Subject: [PATCH 59/59] WIP/9000: arm64: rockchip: rk3288: disable VPU iommu

Due to bug in either hantro, iommu or general dma management drivers for ARMv7
buffers are not correctly flushed when being re-used.

To work around that, disable VPU iommu and increase cma memory size to be
sufficient for 4K H.264 decoding.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
---
 arch/arm/boot/dts/rockchip/rk3288.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/rockchip/rk3288.dtsi b/arch/arm/boot/dts/rockchip/rk3288.dtsi
index 6f363eb14185..3af2cd4f78c3 100644
--- a/arch/arm/boot/dts/rockchip/rk3288.dtsi
+++ b/arch/arm/boot/dts/rockchip/rk3288.dtsi
@@ -1294,6 +1294,7 @@ vpu_mmu: iommu@ff9a0800 {
 		clock-names = "aclk", "iface";
 		#iommu-cells = <0>;
 		power-domains = <&power RK3288_PD_VIDEO>;
+		status = "disabled";
 	};
 
 	hevc: hevc@ff9c0000 {
-- 
2.34.1

